tool/README.md

```diff
@@ -28,24 +27,56 @@ Running each tool without any options (e.g. `./awslc_bm`) will run all available
   "16,256,1350,8192,16384)
 * `-json` has the tool print the output of each benchmark in JSON format.
 
 For example, `./awslc_bm -filter AES -timeout 10 -chunks 16, 256, -json` will run all AES-related tests with input sizes of 16 and 256 for 10 seconds, and output the results in JSON format.
 
-## Setup
-In order to build the above-mentioned benchmarking tools, absolute paths to each libaries' install location must be provided via compiler flags.
-
-### Compiler Flags
-|  Tool Name  |  Compiler Flag  |
-| ------------- | ------------- |
- | bssl speed | (none) | 
-| awslc_bm | -DAWSLC_INSTALL_DIR |
-| bssl_bm | -DBORINGSSL_INSTALL_DIR |
-| ossl_1_0_bm | -DOPENSSL_1_0_INSTALL_DIR |
-| ossl_1_1_bm | -DOPENSSL_1_1_INSTALL_DIR |
-| ossl_3_0_bm | -DOPENSSL_3_0_INSTALL_DIR |
+## Comparison Setup
+The AWS-LC benchmark supports building and running with other common libcryptos.
+Build and install the other libcryptos you would like to test with locally, for 
+example building AWS-LC 2022 FIPS branch and OpenSSL 3.3 to compare with AWS-LC
+main branch:
+
+```bash
+git clone -b openssl-3.3 --depth 1 https://github.com/openssl/openssl.git ~/openssl-3.3-src
+(cd ~/openssl-3.3-src && ./config --prefix="${HOME}/openssl-3.3-install" --openssldir="${HOME}/openssl-3.3-install" && make -j && make install_sw)
+git clone -b fips-2022-11-02 --depth 1 https://github.com/aws/aws-lc.git ~/aws-lc-fips-src
+(cd ~/aws-lc-fips-src && cmake -DCMAKE_INSTALL_PREFIX="${HOME}/aws-lc-fips-install" -DCMAKE_BUILD_TYPE=Release -DFIPS=1 && make -j install)
+```
+
+To build the main branch speed.cc against other libraries pass in the
+BENCHMARK_LIBS option when running CMake. BENCHMARK_LIBS is a list of tuples,
+the format is `executable_name:install_path`. `executable_name` is the name for
+the benchmark that AWS-LC will build with whatever library is in `install_path`.  
+Multiple libraries can be specified with a semicolon between them:
+`executable1_name:executable1_install_path;executable1_name:executable1_install_path;`
+
+To build AWS-LC main speed.cc against the two previously built libcrypto libraries
+(AWS-LC FIPS 2022 and OpenSSL 3.3):
+```
+git clone -b main --depth 1 https://github.com/aws/aws-lc.git ~/aws-lc-main-src
+(cd ~/aws-lc-main-src && cmake -DCMAKE_BUILD_TYPE=Release -DBENCHMARK_LIBS="\
+aws-lc-fips-2022:${HOME}/aws-lc-fips-install;\
+openssl-3-3:${HOME}/openssl-3.3-install;" \
+&& make -j)
+```
+
+This will build 3 relevant binaries:
+* `~/aws-lc-main-src/tool/bssl` is the complete tool build with the main branch of code, `speed` is required to run the benchmark 
+* `~/aws-lc-main-src/tool/aws-lc-fips-2022` is the main branch of speed.cc built with the AWS-LC FIPS 2022 install
+* `~/aws-lc-main-src/tool/openssl-3-3` is the main branch of speed.cc built with the OpenSSL 3.3 install
+
+Not all benchmarks will be available with all libraries, for example OpenSSL 3.3
+does not support ML-KEM.
+
+```
+~/aws-lc-main-src/tool/bssl speed -filter P-256
```

Comment:
The table at the top of the page seems confusing. It's just the names between the brackets that are filter names, but not "EVP ECDH", "ECDH", "Generate" and "ECMUL"? Or do those work as well?

Reply:
Those work as well.